src/sql/workbench/api/common/extHostObjectExplorer.ts

```diff
@@ -7,15 +7,39 @@ import { IMainContext } from 'vs/workbench/api/common/extHost.protocol';
 import { ExtHostObjectExplorerShape, SqlMainContext, MainThreadObjectExplorerShape } from 'sql/workbench/api/common/sqlExtHost.protocol';
 import * as azdata from 'azdata';
 import * as vscode from 'vscode';
+import { ExtHostCommands } from 'vs/workbench/api/common/extHostCommands';
+import { TreeViewItemHandleArg } from 'sql/workbench/common/views';
 
 export class ExtHostObjectExplorer implements ExtHostObjectExplorerShape {
 
 	private _proxy: MainThreadObjectExplorerShape;
 
 	constructor(
-		mainContext: IMainContext
+		mainContext: IMainContext,
+		commands: ExtHostCommands
 	) {
 		this._proxy = mainContext.getProxy(SqlMainContext.MainThreadObjectExplorer);
+
+		function isDataExplorerTreeViewItemHandleArg(arg: any): boolean {
+			return arg?.$treeItem?.payload;
+		}
+
+		function convertDataExplorerArgument(arg: TreeViewItemHandleArg): any {
+			return <azdata.ObjectExplorerContext>{
+				connectionProfile: arg.$treeItem.payload,
+				isConnectionNode: arg.$treeItem.type === 'Server',
```

Comment:
Should we check for `Database` here too for DB connections?

Reply:
Thought that might be an issue with Database nodes from the Databases folder, but those don't have a treeItem.type (instead the NodeInfo type is databases for those) so yeah you're correct.